rust_binding/src/error.rs

```diff
@@ -0,0 +1,403 @@
+            ErrorKind::UnSupportedSearchKey(_) =>
+                "Specified search key is not supported",
+            ErrorKind::EmptyAccessGroup(_) =>
+                "Refused to mask volume to empty access group",
+            ErrorKind::PoolNotReady(_) =>
+                "Pool is not ready for specified action",
+            ErrorKind::DiskNotFree(_) =>
+                "Disk is not free for specified action",
+        }
+    }
+}
+
+impl fmt::Display for LsmError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{}", match self.kind {
+            ErrorKind::LibBug(ref x) => x,
+            ErrorKind::PluginBug(ref x) => x,
+            ErrorKind::TimeOut(ref x) => x,
+            ErrorKind::DaemonNotRunning(ref x) => x,
+            ErrorKind::PermissionDenied(ref x) => x,
+            ErrorKind::NameConflict(ref x) => x,
+            ErrorKind::ExistsInitiator(ref x) => x,
+            ErrorKind::InvalidArgument(ref x) => x,
+            ErrorKind::NoStateChange(ref x) => x,
+            ErrorKind::NetworkConRefused(ref x) => x,
+            ErrorKind::NetworkHostDown(ref x) => x,
+            ErrorKind::NetworkError(ref x) => x,
+            ErrorKind::NoMemory(ref x) => x,
+            ErrorKind::NoSupport(ref x) => x,
+            ErrorKind::IsMasked(ref x) => x,
+            ErrorKind::HasChildDependency(ref x) => x,
+            ErrorKind::NotFoundAccessGroup(ref x) => x,
+            ErrorKind::NotFoundFs(ref x) => x,
+            ErrorKind::NotFoundJob(ref x) => x,
+            ErrorKind::NotFoundPool(ref x) => x,
+            ErrorKind::NotFoundFsSnapshot(ref x) => x,
+            ErrorKind::NotFoundVolume(ref x) => x,
+            ErrorKind::NotFoundNfsExport(ref x) => x,
+            ErrorKind::NotFoundSystem(ref x) => x,
+            ErrorKind::NotFoundDisk(ref x) => x,
+            ErrorKind::NotLicensed(ref x) => x,
+            ErrorKind::NoSupportOnlineChange(ref x) => x,
+            ErrorKind::NoSupportOfflineChange(ref x) => x,
+            ErrorKind::PluginAuthFailed(ref x) => x,
+            ErrorKind::PluginIpcFail(ref x) => x,
+            ErrorKind::PluginSocketPermission(ref x) => x,
+            ErrorKind::PluginNotExist(ref x) => x,
+            ErrorKind::NoEnoughSpace(ref x) => x,
+            ErrorKind::TransportCommunication(ref x) => x,
+            ErrorKind::TransportSerialization(ref x) => x,
+            ErrorKind::TransportInvalidArg(ref x) => x,
+            ErrorKind::LastInitInAccessGroup(ref x) => x,
+            ErrorKind::UnSupportedSearchKey(ref x) => x,
+            ErrorKind::EmptyAccessGroup(ref x) => x,
+            ErrorKind::PoolNotReady(ref x) => x,
+            ErrorKind::DiskNotFree(ref x) => x,
+        })
+    }
+}
+
+impl From<ErrorKind> for LsmError {
+    fn from(ek: ErrorKind) -> LsmError {
+        LsmError{kind: ek}
+    }
+}
+
+pub type Result<T> = result::Result<T, LsmError>;
```

Comment:
@cathay4t @mulkieran Is it a good idea to create the new type with the same name?  I know error chain does this, but...

Reply:
Error chain cannot handle 42+ errors we needed:
https://github.com/rust-lang-nursery/error-chain/issues/117

Reply:
Oh. Just save some typing. not a big deal.